test/fixtures/section-heading-first-fail.html

```diff
@@ -0,0 +1,3 @@
+<section>
+    <p>Hello, world.</p>
+</section>
```

Comment:
The library was written with either/or semantics. I added a new function that checks for "each", which is when a fixed contains a passing and a failing example. Output is as follows (last check):

```
      ✓ nav on nav-aria-label-pass.html, should return true
      ✓ nav on nav-aria-label-value-fail.html, should return false
    nav-aria-label-present
      ✓ nav on nav-aria-label-pass.html, should return true
      ✓ nav on nav-aria-label-present-fail.html, should return false
    link-href
      ✓ a on link-href-pass.html, should return true
      ✓ a on link-href-present-fail.html, should return false
      ✓ a#pound-sign on link-href-value-fail.html, should return false
      ✓ a#empty-href on link-href-value-fail.html, should return false
    link-href-internal
      ✓ a on link-href-pass.html, should return true
      ✓ a on link-href-present-fail.html, should return true
      ✓ a#missing-target on link-href-value-fail.html, should return false
      ✓ a#unfocusable-target on link-href-value-fail.html, should return false
    icon-aria-hidden
      ✓ i.fa on icon-aria-hidden-pass.html, should return true
      ✓ i.icon on icon-aria-hidden-pass.html, should return true
      ✓ span.fa on icon-aria-hidden-pass.html, should return true
      ✓ span.icon on icon-aria-hidden-pass.html, should return true
      ✓ i.fa on icon-aria-hidden-fail.html, should return false
      ✓ i.icon on icon-aria-hidden-fail.html, should return false
      ✓ span.fa on icon-aria-hidden-fail.html, should return false
      ✓ span.icon on icon-aria-hidden-fail.html, should return false
    section-heading-first-child
      ✓ section on section-heading-first-fail.html, should return false
      ✓ section on section-heading-first-pass.html, should return true

  Integration Spec
    ✓ rules can be configured with axe-core
    ✓ checks can be configured with axe-core
    ✓ does not contain unicode chars

  Rules Spec
    skip-link
      ✓ skip-link-fail.html, should fail
      ✓ skip-link-pass.html, should pass
    link-href
      ✓ link-href-value-fail.html, should fail
      ✓ link-href-pass.html, should pass
    nav-aria-label
      ✓ nav-aria-label-present-fail.html, should fail
      ✓ nav-aria-label-value-fail.html, should fail
      ✓ nav-aria-label-pass.html, should pass
    icon-aria-hidden
      ✓ icon-aria-hidden-pass.html, should pass
      ✓ icon-aria-hidden-fail.html, should fail
    description
      ✓ media.html, should not run
    caption
      ✓ media.html, should not run
    section
      ✓ section-heading-first-fail.html, should fail
      ✓ section-heading-first-pass.html, should pass
      ✓ section-heading-first-fail-pass.html, should pass and fail

Firefox 46.0.0 (Mac OS X 10.11.0): Executed 39 of 39 SUCCESS (0.548 secs / 0.507 secs)
TOTAL: 39 SUCCESS
```

I have to admit I don't see the value in having a test with passes and failures, since we could just include any additional incorrect markup to the "fail" file and any additional correct markup to the "pass" file.